apps/src/labs/projects/ProjectManager.ts

```diff
@@ -82,6 +81,14 @@ export default class ProjectManager {
     return this.sourceChanged(project) || this.channelChanged(project);
   }
 
+  // Shut down this project manager. All we do here is clear any existing
+  // timeouts.
+  destroy(): void {
+    if (this.timeoutId) {
+      window.clearTimeout(this.timeoutId);
+    }
```

Comment:
Not really necessary, but we could clear `this.timeoutId` just in case somebody calls `destroy` more than once?